test cases/common/15 mixed pch/meson.build

```diff
@@ -3,4 +3,4 @@ project('mixed C and C++ pch test', 'cpp', 'c')
 
 exe = executable('prog', 'main.cc', 'func.c',
 c_pch : ['pch/func.h', 'pch/func_pch.c'],
-cpp_pch : ['pch/main_pch.cc', 'pch/main.h'])
+cpp_pch : ['pch/main_pch.cc', 'pch/main.hh'])
```

Comment:
Why the renaming? We should be able to support `.h` in C++ because that is what most people use.